File: examples/example-java/src/main/java/nl/jandt/block/example/FlightCommand.java
package nl.jandt.block.example;

import net.minestom.server.entity.Player;
import nl.jandt.blocky.engine.impl.command.BlockyCommand;
import nl.jandt.blocky.engine.core.server.Server;

public class FlightCommand extends BlockyCommand {
    public FlightCommand(Server server) {
        super(server, "fly");

        addSyntax((sender, context) -> {
            if (sender instanceof Player player) {
                final var playerObject = getPlayerObject(player);
                assert playerObject != null;

                final var flightTrait = playerObject.tryGetTrait(PlayerAllowFlight.class);

                if (flightTrait.isPresent()) {
                    playerObject.removeTrait(flightTrait.get());
                    sender.sendMessage("You can no longer fly!");
                } else {
                    playerObject.tryAddTrait(PlayerAllowFlight.class);
                    sender.sendMessage("You can now fly!");
                }
            }
        });
    }
}

File: examples/example-kotlin/src/main/kotlin/FlightCommand.kt
package nl.jandt.blocky

import net.minestom.server.entity.Player
import nl.jandt.blocky.engine.core.server.Server
import nl.jandt.blocky.engine.impl.command.BlockyCommand

class FlightCommand(server: Server) : BlockyCommand(server, "fly") {
    init {
        addSyntax({ sender, _ ->
            if (sender !is Player) return@addSyntax

            val playerObject = getPlayerObject(sender)
            checkNotNull(playerObject)

            val flightTrait: PlayerAllowFlight? = playerObject.getTrait(PlayerAllowFlight::class.java)

            if (flightTrait != null) {
                playerObject.removeTrait(flightTrait)
                sender.sendMessage("You can no longer fly!")
            } else {
                playerObject.addTrait(PlayerAllowFlight::class.java)
                sender.sendMessage("You can now fly!")
            }
        })
    }
}
